src/Raven.Server/Documents/Queries/QueryMetadata.cs

```diff
@@ -2317,14 +2319,24 @@ public void HandleSpatial(string methodName, List<QueryExpression> arguments, bo
                             if (spatialExpression.Arguments.Count != 2)
                                 throw new InvalidQueryException($"Method {methodName}() expects first argument to be a point() method with 2 arguments", QueryText, parameters);
 
-                            var latitude = _metadata.ExtractFieldNameFromArgument(spatialExpression.Arguments[0], withoutAlias, "point", parameters, QueryText).Value;
-                            var longitude = _metadata.ExtractFieldNameFromArgument(spatialExpression.Arguments[1], withoutAlias, "point", parameters, QueryText).Value;
+                            var latitudePropertyPath = _metadata.ExtractFieldNameFromArgument(spatialExpression.Arguments[0], withoutAlias, "point", parameters, QueryText).Value;
+                            var longitudePropertyPath = _metadata.ExtractFieldNameFromArgument(spatialExpression.Arguments[1], withoutAlias, "point", parameters, QueryText).Value;
 
                             fieldOptions = new AutoSpatialOptions(AutoSpatialOptions.AutoSpatialMethodType.Point, new List<string>
                             {
-                                latitude,
-                                longitude
+                                latitudePropertyPath,
+                                longitudePropertyPath
                             });
+
+                            if (_metadata.SelectFields != null)
+                            {
+                                latitudePropertyPath = GetPathWithAliasName(latitudePropertyPath);
+                                longitudePropertyPath = GetPathWithAliasName(longitudePropertyPath);
+                            }
+                            
+                            _metadata.SpatialResults ??= new List<SpatialProperty>();
+                            _metadata.SpatialResults.Add(new SpatialProperty(latitudePropertyPath, longitudePropertyPath));
```

Comment:
This is only needed for studio. Technically we could probably do it only if it's explicitly requested by the studio. Although query metadata are cached so I think it's okay to have this information always in metadata - it will be calculated just once per query.